airflow-core/src/airflow/configuration.py

```diff
@@ -202,6 +203,28 @@ def retrieve_configuration_description(
     return base_configuration_description
 
 
+class WeightRule(str, Enum):
```

Comment:
What was you thinking for putting this in configuration.py? On seeing this it looks like a very odd location to put this enum

Reply:
It purely seems to be based on the usage here. Maybe a new module inside `airflow.operators` sounds better?

Reply:
It is only being used [here](https://github.com/astronomer/airflow/blob/fa038ddd2e92e0a304dd7fee2d8b55e7f126f7f6/airflow-core/src/airflow/configuration.py#L438) 

I did not find better place for this as it was not being used elsewhere

Reply:
https://github.com/astronomer/airflow/blob/fa038ddd2e92e0a304dd7fee2d8b55e7f126f7f6/airflow-core/src/airflow/models/taskinstance.py#L547-L549 -- that's where it's actually used in the server side of things.

Reply:
However I'm not sure if weight_rule has actually been working correctly since https://github.com/astronomer/airflow/commit/9855515becfb85a06dc9a64e50b0c5b4e5a6597c landed. Iiinteresting.